[Bug]: Errors NU3028 and NU3037 when restoring NuGet packages on FreeBSD

### NuGet Product Used

dotnet.exe

### Product Version

.NET SDK (reflecting any global.json):  Version:   6.0.100-dev  Commit:    2f6dac4342  Runtime Environment:  OS Name:     FreeBSD  OS Version:  12  OS Platform: FreeBSD  RID:         freebsd.12-x64  Base Path:   /usr/local/share/dotnet/sdk/6.0.100-dev/  Host (useful for support):   Version: 6.0.0   Commit:  N/A  .NET SDKs installed:   6.0.100-dev [/usr/local/share/dotnet/sdk]  .NET runtimes installed:   Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]   Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]  To install additional .NET runtimes or SDKs:   https://aka.ms/dotnet-download

### Worked before?

It worked until the root certificate of the corresponding timestamp provider chain was removed: https://github.com/NuGet/Announcements/issues/56

### Impact

It's more difficult to complete my work

### Repro Steps & Context

This issue has already been fixed by [#3979](https://github.com/NuGet/NuGet.Client/pull/3979) for Linux and MacOS but not for FreeBSD, because [RuntimeEnvironmentHelper.IsLinux](https://github.com/NuGet/NuGet.Client/blob/1fdb7892fa61aae87d6ea67d522b42eca5f282a3/src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs#L139-L158) returns `false` if `IS_CORECLR` is defined. According to [common.targets](https://github.com/NuGet/NuGet.Client/blob/1fdb7892fa61aae87d6ea67d522b42eca5f282a3/build/common.targets#L9-L12) `IS_CORECLR` is defined for all target frameworks starting with `netcoreapp`, `netstandard` or `net6`, and therefore the code that causes this issue is included by a NuGet package in the `dotnet` executable.

### Verbose Logs

```shell
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json': 
  Signature type: Repository
  Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json':   Subject Name: CN=NuGet.org Repository by Microsoft, O=NuGet.org Repository by Microsoft, L=Redmond, S=Washington, C=US
  Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json':   SHA256 hash: 0E5F38F57DC1BCC806D8494F4F90FBCEDD988B46760709CBEEC6F4219AA6157D
  Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json':   Valid from: 04/10/2018 00:00:00 to 04/14/2021 12:00:00
/usr/home/Administrator/ConsoleApp/ConsoleApp.csproj : error NU3037: Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json': The repository primary signature validity period has expired.
/usr/home/Administrator/ConsoleApp/ConsoleApp.csproj : error NU3028: Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json': The repository primary signature's timestamping certificate is not trusted by the trust provider.
  Failed to restore /usr/home/Administrator/ConsoleApp/ConsoleApp.csproj (in 1.37 sec).

Build FAILED.

/usr/home/Administrator/ConsoleApp/ConsoleApp.csproj : error NU3037: Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json': The repository primary signature validity period has expired.
/usr/home/Administrator/ConsoleApp/ConsoleApp.csproj : error NU3028: Package 'System.Runtime 4.0.0' from source 'https://api.nuget.org/v3/index.json': The repository primary signature's timestamping certificate is not trusted by the trust provider.
    0 Warning(s)
    2 Error(s)
```

--- a/src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs
+++ b/src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs
@@ -150,6 +150,13 @@ namespace NuGet.Common
                 return true;
             }
 
+            // The OSPlatform.FreeBSD property only exists in .NET Core 3.1 and higher, whereas this project is also
+            // compiled for .NET Standard and .NET Framework, where an OSPlatform for FreeBSD must be created manually
+            if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Create("FREEBSD")))
+            {
+                return true;
+            }
+
             return false;
 #else
             var platform = (int)Environment.OSVersion.Platform;

